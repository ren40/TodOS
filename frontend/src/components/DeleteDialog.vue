<template>
  <v-dialog v-model="dialog" max-width="290" @click:outside="$emit('delete', false)">
    <v-card>
      <v-card-title class="text-h5">
      <slot name="title"></slot>
      </v-card-title>

      <v-card-text>
        <slot name="description"></slot>
      </v-card-text>

      <v-card-actions>
        <v-spacer></v-spacer>

        <v-btn color="green darken-1" text @click="$emit('delete', false)">
          Disagree
        </v-btn>

        <v-btn color="error darken-1" text @click="$emit('delete', true)">
          Agree
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  name: "DeleteDialog",
  props: ["dialog"],
};
</script>

<style>
</style>