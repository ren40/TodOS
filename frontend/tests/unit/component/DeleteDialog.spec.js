import DeleteDialog from "@/components/DeleteDialog";
import { createLocalVue, mount } from "@vue/test-utils";
import Vuetify from "vuetify";
const localVue = createLocalVue();

describe("DeleteDialog.vue", () => {
  let wrapper;
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("Testing the rendering of the DeleteDialog component", () => {
    wrapper = mount(DeleteDialog, {
      localVue,
      vuetify,
      propsData: {
        dialog: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
