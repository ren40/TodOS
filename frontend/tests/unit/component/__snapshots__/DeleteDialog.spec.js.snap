// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteDialog.vue  1`] = `<div role="dialog" class="v-dialog__container"></div>`;
